test/extended/builds/forcepull.go

```diff
@@ -53,6 +59,13 @@ func doTest(bldPrefix, debugStr string, same bool, oc *exutil.CLI) {
 	// dump tags/hexids for debug
 	_, err = exutil.DumpAndReturnTagging(tags)
 	o.ExpectWithOffset(1, err).NotTo(o.HaveOccurred())
+
+	if bldPrefix == buildPrefixFC || bldPrefix == buildPrefixTC {
+		defer func() {
+			err = oc.AsAdmin().Run("adm").Args("policy", "remove-cluster-role-from-user", "system:build-strategy-custom", oc.Username()).Execute()
+			o.Expect(err).NotTo(o.HaveOccurred())
+		}()
```

Comment:
you need to setup the defer immediately after adding the role, the defer has to be in place before you start running the actual test steps.